lib/gameState/GameStatistics.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * GameSTatistics.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#pragma once
+
+#include "../GameConstants.h"
+#include "../ResourceSet.h"
+
+VCMI_LIB_NAMESPACE_BEGIN
+
+struct PlayerState;
+class CGameState;
+class CGHeroInstance;
+class CGMine;
+
+struct DLL_LINKAGE StatisticDataSetEntry
+{
+	std::string map;
+	time_t timestamp;
+    int day;
+    PlayerColor player;
+	TeamID team;
+	bool isHuman;
+	EPlayerStatus status;
+	TResources resources;
+	int numberHeroes;
+	int numberTowns;
+	int numberArtifacts;
+	si64 armyStrength;
+	int income;
+	double mapVisitedRatio;
```

Comment:
1. Explored, not visited.
2. Float is precise enough even for G+U maps
```suggestion
	float mapExploredRatio;
```